### PR TITLE
replace whitespaces with underlines in feature name

### PR DESCRIPTION
**Pull request by wxchan**
#379

### CODE REVIEW

File: include/LightGBM/dataset.h [hunk 1]
```diff
@@ -495,10 +495,21 @@ class Dataset {
 
   inline void set_feature_names(const std::vector<std::string>& feature_names) {
     if (feature_names.size() != static_cast<size_t>(num_total_features_)) {
-      Log::Warning("size of feature_names error, should equal with total number of features");
+      Log::Warning("Size of feature_names error, should equal with total number of features");
```
Comment (wxchan) [inline include/LightGBM/dataset.h]: why it's not error here? @guolinke

Comment (guolinke) [inline include/LightGBM/dataset.h]: why it will have error ?

Comment (wxchan) [inline include/LightGBM/dataset.h]: I changed `dataset->feature_names_ = feature_names_;` to `dataset->set_feature_names(feature_names_);`. so if `feature_names.size() != static_cast<size_t>(num_total_features_)`, `dataset->feature_names_` will be null. I didn't change this yet, you can review rest of pr first.

Comment (guolinke) [inline include/LightGBM/dataset.h]: ok. You can change it to ```Log::Fatal``` .

File: src/io/dataset_loader.cpp [hunk 2]
```diff
@@ -28,7 +28,7 @@ void DatasetLoader::SetHeader(const char* filename) {
     // get column names
     if (io_config_.has_header) {
       std::string first_line = text_reader.first_line();
-      feature_names_ = Common::Split(first_line.c_str(), "\t ,");
+      feature_names_ = Common::Split(first_line.c_str(), "\t,");
```
Comment (guolinke) [inline src/io/dataset_loader.cpp]: why not enable split by space here ?

Comment (wxchan) [inline src/io/dataset_loader.cpp]: I think the data format is either csv (split by ',') or tsv (split by '\t'), is whitespace allowed for splitting the columns? whitespace can appear in feature name.

Comment (guolinke) [inline src/io/dataset_loader.cpp]: ok

**Pull request merged**